From b4d210aa01b4b233e7ba348093b636a03fdf67ba Mon Sep 17 00:00:00 2001
From: Dan Prince <dprince@redhat.com>
Date: Thu, 20 Jun 2013 09:34:28 -0400
Subject: [PATCH] Add support for x86_64 boot-stack VMs.

Adds a new --arch option to configure-seed-vm. This option is
passed into the seed.xml libvirt template to control the arch
being used.

Also, Updates the scripts/boot-elements script so that it
passes the correct architecture when creating the seed VM.

With this change I'm able to create x86_64 bootstrap VMs...
---
 bootstrap/configure-seed-vm | 5 ++++-
 bootstrap/seed.xml          | 2 +-
 scripts/boot-elements       | 9 ++++++---
 3 files changed, 11 insertions(+), 5 deletions(-)

diff --git a/bootstrap/configure-seed-vm b/bootstrap/configure-seed-vm
index ad36f8f..2d3f377 100755
--- a/bootstrap/configure-seed-vm
+++ b/bootstrap/configure-seed-vm
@@ -18,6 +18,8 @@ def main():
         help='The interface which bare metal nodes will be connected to.')
     parser.add_argument('--engine', default='kvm',
         help='The virtualization engine to use')
+    parser.add_argument('--arch', default='i686',
+        help='The architecture to use')
     args = parser.parse_args()
     with file(basedir + '/seed.xml', 'rb') as f:
         source_template = f.read()
@@ -29,7 +31,8 @@ def main():
         'name': args.name,
         'imagefile': imagefile,
         'bmbridge': args.baremetal_interface,
-        'engine': args.engine
+        'engine': args.engine,
+        'arch': args.arch
         }
     if args.image is not None:
         params['imagefile'] = args.image
diff --git a/bootstrap/seed.xml b/bootstrap/seed.xml
index 677a28f..c7e2760 100644
--- a/bootstrap/seed.xml
+++ b/bootstrap/seed.xml
@@ -3,7 +3,7 @@
   <memory unit='KiB'>1572864</memory>
   <vcpu>2</vcpu>
   <os>
-    <type arch='i686' machine='pc-1.0'>hvm</type>
+    <type arch='%(arch)s' machine='pc-1.0'>hvm</type>
     <boot dev='hd'/>
   </os>
   <features>
diff --git a/scripts/boot-elements b/scripts/boot-elements
index dc0cc46..c17cf65 100755
--- a/scripts/boot-elements
+++ b/scripts/boot-elements
@@ -72,9 +72,11 @@ if [ ! "$EXTRA_ELEMENTS" ]; then
     show_options
 fi
 
+SEED_ARCH=
+
 case $ARCH in
-    i386) ;;
-    amd64) ;;
+    i386) SEED_ARCH='i686'; ;;
+    amd64) SEED_ARCH='x86_64'; ;;
     *) echo "Unsupported arch $ARCH!" ; exit 1 ;;
 esac
 
@@ -105,7 +107,8 @@ sudo mv /tmp/$IMAGE_NAME.qcow2 /var/lib/libvirt/images/
 $SCRIPT_HOME/../bootstrap/configure-seed-vm \
            --name $IMAGE_NAME \
            --image /var/lib/libvirt/images/$IMAGE_NAME.qcow2 \
-           --engine $ENGINE
+           --engine $ENGINE \
+           --arch $SEED_ARCH
 
 VM_IP=''
 function poll_vm {
-- 
1.8.1.4

